src/preview.cpp

```diff
@@ -545,7 +546,7 @@ namespace {
             }
         }
 
-        if (offset_ + size_ > static_cast<uint32_t>(image_.io().size())) return;
+        if (Safe::add(offset_, size_) > static_cast<uint32_t>(image_.io().size())) return;
```

Comment:
[minor] please move the `return` to a new line.

Reply:
Done. Totally missed the return there, was too far to the right.